zerver/views/auth.py

```diff
@@ -1040,3 +1041,21 @@ def saml_sp_metadata(request: HttpRequest, **kwargs: Any) -> HttpResponse:  # no
                             content_type='text/xml')
 
     return HttpResponseServerError(content=', '.join(errors))
+
+def config_error_view(request: HttpRequest, error_category_name: str) -> HttpResponse:
+    def config_error(context: object) -> HttpResponse:
+        return TemplateView.as_view(template_name='zerver/config_error.html',
+                                    extra_context=context)
+    contexts = {
+        'google': {'social_backend_name': 'google', 'has_markdown_file': True},
+        'github': {'social_backend_name': 'github', 'has_markdown_file': True},
+        'gitlab': {'social_backend_name': 'gitlab', 'has_markdown_file': True},
+        'ldap': {'error_name': 'ldap_error_realm_is_none'},
+        'dev': {'error_name': 'dev_not_supported_error'},
+        'saml': {'social_backend_name': 'saml'},
+        'smtp': {'error_name': 'smtp_error'},
+        'backend_disabled': {'error_name': 'remoteuser_error_backend_disabled'},
+        'remote_user_header_missing': {'error_name': 'remoteuser_error_remote_user_header_missing'}
+    }
+
+    return config_error(contexts[error_category_name])(request)
```

Comment:
I think it's simpler to just do something like
```
    context = contexts[error_category_name]
    return TemplateView.as_view(template_name='zerver/config_error.html',
                                extra_context=context)(request)
```
and avoid the nested function thing?

Reply:
It'll be clearer, changed it. The first version of nested function had some other things. while changing it, didn't think it's no different than just using `TemplateView.as_view....`. Anyways thanks!